src/datasets/utils/filelock.py

```diff
@@ -377,7 +375,7 @@ def _release(self):
         os.close(fd)
 
         try:
-            os.remove(self._lock_file)
+            os.remove("\\\\?\\" + os.path.abspath(self._lock_file))
```

Comment:
A tiny nit. `os.path.abspath(os.path.expanduser(os.path.expandvars(self._lock_file)))` is a more robust conversion.